### PR TITLE
Migrate from build dependency on jcenter (EXPOSUREAPP-7437)

### PR DESCRIPTION
**Pull request by mtwalli**
- mockK, detekt, screengrab are using indirect dependencies that uses `jcenter()`
- New version of most libraries already migrated from `jcenter` to `mavenCentral` except Screengrab  

- Mostly no changes in the App code - just test classes 
- Refactored Screenshots a bit by moving the SystemUiRule into the super class
- Note: I deleted mock in Home Screen Test, it is not related to this change ,but the mocks are not required anymore since we moved away from fastlane

### CODE REVIEW

**Comment by mtwalli** [comment]
Fixes: https://github.com/corona-warn-app/cwa-app-android/issues/3264

File: Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/bugreporting/DebugLogFragmentTest.kt [hunk 1]
```diff
@@ -38,13 +35,6 @@ class DebugLogFragmentTest : BaseUITest() {
     private lateinit var inactiveViewModel: DebugLogViewModel
     private lateinit var activeViewModel: DebugLogViewModel
 
-    @Rule
-    @JvmField
-    val localeTestRule = LocaleTestRule()
```
Comment (mtwalli) [inline Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/bugreporting/DebugLogFragmentTest.kt:43]: Part of Screengrab

File: Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/bugreporting/DebugLogFragmentTest.kt [hunk 2]
```diff
@@ -38,13 +35,6 @@ class DebugLogFragmentTest : BaseUITest() {
...
-
-    @get:Rule
-    val systemUIDemoModeRule = SystemUIDemoModeRule()
```
Comment (mtwalli) [inline Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/bugreporting/DebugLogFragmentTest.kt:46]: Moved to super class

File: Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/contactdiary/ContactDiaryDayFragmentTest.kt [hunk 3]
```diff
@@ -69,7 +58,7 @@ class ContactDiaryDayFragmentTest : BaseUITest() {
 
         every { viewModel.uiState } returns MutableLiveData(
             ContactDiaryDayViewModel.UIState(
-                dayText = { LocalDate.now().toFormattedDay(LocaleUtil.getTestLocale() ?: Locale.getDefault()) },
```
Comment (mtwalli) [inline Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/contactdiary/ContactDiaryDayFragmentTest.kt:72]: LocaleUtil is also part of Screengrab

File: Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/contactdiary/ContactDiaryOverviewFragmentTest.kt [hunk 4]
```diff
@@ -43,33 +33,11 @@ import testhelpers.takeScreenshot
...
-                taskController = taskController,
-                dispatcherProvider = TestDispatcherProvider(),
-                contactDiaryRepository = contactDiaryRepository,
-                riskLevelStorage = riskLevelStorage,
-                timeStamper = timeStamper,
-                checkInRepository = checkInRepository,
-                exporter = exporter
-            )
-        )
-
```
Comment (mtwalli) [inline Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/contactdiary/ContactDiaryOverviewFragmentTest.kt:72]: Not needed anymore , we can just mock the ViewModel itself.

File: Corona-Warn-App/src/androidTest/java/testhelpers/BaseUITest.kt [hunk 5]
```diff
@@ -15,6 +15,9 @@ abstract class BaseUITest : BaseTestInstrumentation() {
         Manifest.permission.WRITE_EXTERNAL_STORAGE
     )
 
+    @get:Rule
+    val systemUIDemoModeRule = SystemUIDemoModeRule()
+
```
Comment (mtwalli) [inline Corona-Warn-App/src/androidTest/java/testhelpers/BaseUITest.kt:20]: Super class has the rule by default , so it is no needed to do that for subclasses

File: Corona-Warn-App/src/androidTest/java/testhelpers/ScreenShotter.kt [hunk 6]
```diff
@@ -26,11 +23,8 @@ inline fun <reified T> takeScreenshot(suffix: String = "", delay: Long = SCREENS
     Thread.sleep(delay)
     val simpleName = T::class.simpleName
     val name = if (suffix.isEmpty()) simpleName else simpleName.plus("_$suffix")
-    Screengrab.screenshot(
-        name,
-        UiAutomatorScreenshotStrategy(),
-        SDCardCallback
-    )
+    val screenshot = InstrumentationRegistry.getInstrumentation().uiAutomation.takeScreenshot()
+    SDCard.screenshotCaptured(name ?: System.currentTimeMillis().toString(), screenshot)
```
Comment (mtwalli) [inline Corona-Warn-App/src/androidTest/java/testhelpers/ScreenShotter.kt:27]: These two lines replace Screengrab screenshots logic  😮

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/appconfig/sources/fallback/DefaultAppConfigSourceTest.kt [hunk 7]
```diff
@@ -82,7 +82,8 @@ class DefaultAppConfigSourceTest : BaseIOTest() {
 
     @Test
     fun `exceptions when getting the default config are rethrown`() = runBlockingTest {
-        every { assetManager.open("default_app_config_android.bin") } throws FileNotFoundException("default_app_config_android.bin does not exist")
+        every { assetManager.open("default_app_config_android.bin") } throws
+            FileNotFoundException("default_app_config_android.bin does not exist")
```
Comment (mtwalli) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/appconfig/sources/fallback/DefaultAppConfigSourceTest.kt:86]: Changes regarding formatting in Test classes are due to `detekt` newer version which now check Test classes too

File: build.gradle [hunk 8]
```diff
@@ -7,7 +7,7 @@ buildscript {
 
     repositories {
         google()
-        jcenter()
+        mavenCentral()
```
Comment (mtwalli) [inline build.gradle:10]: Finally ,this is actually what is supposed to be in the PR ?!

**Review by mtwalli — COMMENTED** [review]
Added some comments to help the reviewers

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/coronatest/qrcode/TestQrCodes.kt [hunk 9]
```diff
@@ -1,25 +1,49 @@
...
-internal val raQrCode6 =
-    "https://s.coronawarn.app?v=1#eyJ0aW1lc3RhbXAiOjE2MTk2MTgzOTcsInNhbHQiOiI3MDcyQzAyMTkyM0ZFOEMzMDVDNTAxQkU5MjQyMjNBQyIsInRlc3RpZCI6IjhmMzA0OTE4LWI2YmMtNDQyOS1iYzhlLWMzMzRkMjdhNTdiNCIsImhhc2giOiI2ZjVhMjJhYzY2ZTc1Y2JiYTE3MTBlN2IxZWMwZTllMDk4NjUyMjY0MWE3NTYyNGY0MGZhMDc4YTZkZjY0ZTVjIiwiZm4iOiJBbmRyZSIsImxuIjoiQmFyZ2VsbGluaSIsImRvYiI6IjE5OTItMTEtMDcifQ=="
-internal val raQrCode7 =
-    "https://s.coronawarn.app?v=1#eyJ0aW1lc3RhbXAiOjE2MTk2MTg0MjMsInNhbHQiOiJEM0Y1MzcyODU2NkMxMDFENjE1MkVCQ0I0OEMxMkFCOCIsInRlc3RpZCI6Ijc5NWIxY2MwLWU2NjQtNGFmZi05NTk3LWU3MTk2ODE4ZGVmYiIsImhhc2giOiI3NmMxMjJiOTlmZWVmZmM5Mjc3MTE2YjUwZGIwZGM1NjI0ZjY5OWFiMzliMDAwOWMwYzg5YmRlMWNjZjM4YmQxIiwiZm4iOiJWaWN0b3JpYSIsImxuIjoiTWFubmVsbGkiLCJkb2IiOiIxOTc4LTA0LTIxIn0="
-internal val raQrCode8 =
-    "https://s.coronawarn.app?v=1#eyJ0aW1lc3RhbXAiOjE2MTk2MjEzMzEsInNhbHQiOiI5NERDMkUyRkNCNTE2NDFFQzA5RkZENjVEMkJEMDg4QiIsInRlc3RpZCI6ImMyMGI5OTgxLWZmNzAtNGRmOC1hYTAyLTUwOTdmMmJkN2YzYyIsImhhc2giOiI1MmVlZjA3YjQwYzU1ODM0OTg3MGQ3YTA2Yzc0OGIwOTAxNGMxMTBlYTkzYjZhNmRhNDhkOWI4NTU0NTU2MzY0In0="
+internal const val raQrCode1 =
```
Comment (d4rken) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/coronatest/qrcode/TestQrCodes.kt:7]: Please just supress the LINT for this class. Linebreaks just make it less readable imho and make it more difficult to compare and copy paste.

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/presencetracing/checkins/qrcode/Base32UrlProvider.kt [hunk 10]
```diff
@@ -76,6 +76,7 @@ class Base32UrlProvider : ArgumentsProvider {
     companion object {
         const val CRYPTOGRAPHIC_SEED = "MTIzNA=="
         const val PUB_KEY =
-            "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEc7DEstcUIRcyk35OYDJ95/hTg3UVhsaDXKT0zK7NhHPXoyzipEnOp3GyNXDVpaPi3cAfQmxeuFMZAIX2+6A5Xg=="
+            "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEc7DEstcUIRcyk35OYDJ95/hTg3UVhsaDXKT" +
```
Comment (d4rken) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/presencetracing/checkins/qrcode/Base32UrlProvider.kt:79]: Please just suppress.

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/util/NoPaddingTestProvider.kt [hunk 11]
```diff
@@ -18,11 +18,13 @@ class NoPaddingTestProvider : ArgumentsProvider {
             ),
             Arguments.of(
                 "die heiße zypernsonne quälte max und victoria ja böse auf dem weg bis zur küste.",
-                "MRUWKIDIMVU4HH3FEB5HS4DFOJXHG33ONZSSA4LVYOSGY5DFEBWWC6BAOVXGIIDWNFRXI33SNFQSA2TBEBRMHNTTMUQGC5LGEBSGK3JAO5SWOIDCNFZSA6TVOIQGXQ54ON2GKLQ"
+                "MRUWKIDIMVU4HH3FEB5HS4DFOJXHG33ONZSSA4LVYOSGY5DFEBWWC6BAOVXGIIDWNFR" +
```
Comment (d4rken) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/util/NoPaddingTestProvider.kt:21]: Single line with suppress please

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/util/NoPaddingTestProvider.kt [hunk 12]
```diff
@@ -18,11 +18,13 @@ class NoPaddingTestProvider : ArgumentsProvider {
...
+                    "XI33SNFQSA2TBEBRMHNTTMUQGC5LGEBSGK3JAO5SWOIDCNFZSA6TVOIQGXQ54ON2GKLQ"
             ),
             Arguments.of(
                 "DIE HEISSE ZYPERNSONNE QUÄLTE MAX UND VICTORIA JA BÖSE AUF DEM WEG BIS ZUR KÜSTE.",
-                "IREUKICIIVEVGU2FEBNFSUCFKJHFGT2OJZCSAUKVYOCEYVCFEBGUCWBAKVHEIICWJFBVIT2SJFASASSBEBBMHFSTIUQECVKGEBCEKTJAK5CUOICCJFJSAWSVKIQEXQ44KNKEKLQ"
+                "IREUKICIIVEVGU2FEBNFSUCFKJHFGT2OJZCSAUKVYOCEYVCFEBGUCWBAKVHEIICWJFBVIT" +
```
Comment (d4rken) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/util/NoPaddingTestProvider.kt:26]: Single line with suppress please

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/util/WithPaddingTestProvider.kt [hunk 13]
```diff
@@ -18,11 +18,13 @@ class WithPaddingTestProvider : ArgumentsProvider {
             ),
             Arguments.of(
                 "die heiße zypernsonne quälte max und victoria ja böse auf dem weg bis zur küste.",
-                "MRUWKIDIMVU4HH3FEB5HS4DFOJXHG33ONZSSA4LVYOSGY5DFEBWWC6BAOVXGIIDWNFRXI33SNFQSA2TBEBRMHNTTMUQGC5LGEBSGK3JAO5SWOIDCNFZSA6TVOIQGXQ54ON2GKLQ="
+                "MRUWKIDIMVU4HH3FEB5HS4DFOJXHG33ONZSSA4LVYOSGY5DFEBWWC6BAOVXGIIDWNFRXI33SNF" +
```
Comment (d4rken) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/util/WithPaddingTestProvider.kt:21]: Single line with suppress please

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/util/WithPaddingTestProvider.kt [hunk 14]
```diff
@@ -18,11 +18,13 @@ class WithPaddingTestProvider : ArgumentsProvider {
...
+                    "QSA2TBEBRMHNTTMUQGC5LGEBSGK3JAO5SWOIDCNFZSA6TVOIQGXQ54ON2GKLQ="
             ),
             Arguments.of(
                 "DIE HEISSE ZYPERNSONNE QUÄLTE MAX UND VICTORIA JA BÖSE AUF DEM WEG BIS ZUR KÜSTE.",
-                "IREUKICIIVEVGU2FEBNFSUCFKJHFGT2OJZCSAUKVYOCEYVCFEBGUCWBAKVHEIICWJFBVIT2SJFASASSBEBBMHFSTIUQECVKGEBCEKTJAK5CUOICCJFJSAWSVKIQEXQ44KNKEKLQ="
+                "IREUKICIIVEVGU2FEBNFSUCFKJHFGT2OJZCSAUKVYOCEYVCFEBGUCWBAKVHEIICWJFBVIT2SJFA" +
```
Comment (d4rken) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/util/WithPaddingTestProvider.kt:26]: Single line with suppress please

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/vaccination/core/VaccinationTestData.kt [hunk 15]
```diff
@@ -16,7 +16,14 @@ class VaccinationTestData @Inject constructor(
 
     // AndreasAstra1.pdf
     val personAVac1QRCodeString =
-        "HC1:6BFOXN*TS0BI\$ZD.P9UOL97O4-2HH77HRM3DSPTLRR+%3KXH9M9ESIGUBA KWML%6S5B9-+P70Q5VC9:BPCNYKMXEE1JAA/CXGG0JK1WL260X638J3-E3ND3DAJ-43TTTO3HK1H3QBCWNZ83UQJ:T0/8F7V0HKN:Q8.HBV+0SZ4GH00T9UKP0T9WC5PF6846A\$Q$76QW6%V98T5\$FQMI5DN9QZ5Y0Q\$UPE%5MZ5*T57ZA\$O7T6LEJOA+MZ55EII-EB1EKC422JBBD0D2K.EJJ14B2MP41WTRZPQEC5L64HX6IAS 8S8FT/MAMXP6QS03L0QIRR97I2HOAXL92L0. KOKG8VG5SI:TU+MMPZ55%PBT1YEGEA7IB65C94JBQ2NLEE:NQ% GC3MXHFLF9OIFN0IZ95LJL80P1FDLW452I8941:HH3M41GTNP8EFUNT$.FTD852IWKP/HLIJL8JF8JF172IMAS EDAHMXFBFBQSKJE72KV\$FHJ%3O%6:XM+1QD+T2/VKKER3L3%1THL7MGY.1S:T:GLOX6OCE7+RWYL3.C-L27WNV0G::M74O%K7C50AAEI4"
+        "HC1:6BFOXN*TS0BI\$ZD.P9UOL97O4-2HH77HRM3DSPTLRR+%3KXH9M9ESIGUBA KWML%6S5B9-" +
```
Comment (d4rken) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/vaccination/core/VaccinationTestData.kt:19]: Single line with suppress please

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/vaccination/core/VaccinationTestData.kt [hunk 16]
```diff
@@ -68,7 +75,15 @@ class VaccinationTestData @Inject constructor(
 
     // AndreasAstra2.pdf
     val personAVac2QRCodeString =
-        "6BFOXN*TS0BI\$ZD.P9UOL97O4-2HH77HRM3DSPTLRR+%3D H9M9ESIGUBA KWMLYX1HXK 0DV:D5VC9:BPCNYKMXEE1JAA/CZIK0JK1WL260X638J3-E3ND3DAJ-43TTTMDF6S8:B73QN VNZ.0K6HYI3CNN96BPHNW*0I85V.499TXY9KK9%OC+G9QJPNF67J6QW67KQ9G66PPM4MLJE+.PDB9L6Q2+PFQ5DB96PP5/P-59A%N+892 7J235II3NJ7PK7SLQMIPUBN9CIZI.EJJ14B2MP41IZRZPQEC5L64HX6IAS 8SAFT/MAMXP6QS03L0QIRR97I2HOAXL92L0. KOKGGVG5SI:TU+MMPZ55%PBT1YEGEA7IB65C94JBQ2NLEE:NQ% GC3MXHFLF9OIFN0IZ95LJL80P1FDLW452I8941:HH3M41GTNP8EFUNT\$.FTD852IWKP/HLIJL8JF8JF172E2JA0K*WDQMPB8T3%KLUSR43M.F\$QBQDR\$VT7V01Y7J0BOZLH+D-QF6MO\$R3%XB+.4QI596GY\$SITJP5BS0DFROC.7B.2RTB*UNYSM$*00HIL+H"
+        "6BFOXN*TS0BI\$ZD.P9UOL97O4-2HH77HRM3DSPTLRR+%3D H9M9ESIGUBA KWMLYX1HXK " +
```
Comment (d4rken) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/vaccination/core/VaccinationTestData.kt:78]: Single line with suppress please

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/vaccination/core/VaccinationTestData.kt [hunk 17]
```diff
@@ -124,7 +139,16 @@ class VaccinationTestData @Inject constructor(
 
     // BorisJohnson1.pdf
     val personBVac1QRCodeString =
-        "HC1:6BFOXN*TS0BI\$ZD.P9UOL97O4-2HH77HRM3DSPTLRR+%3QVH9M9ESIGUBA KWML:SPHXK 0DMYF5VC9:BPCNYKMXEE1JAA/CZIK0JK1WL260X638J3-E3ND3DAJ-43 QTCPFFIJRF3O8H43HX37DUF GFE VMJJYC3SM74E5V.499TXY9KK9+OC+G9QJPNF67J6QW67KQ2G66PPM4MLJE+.PDB9L6Q2+PFQ5DB96PP5/P-59A%N+892 7J235II3NJ7PK7SLQMIPUBN9CIZI.EJJ14B2MP41AZRSEQEC5L64HX6IAS3DS2980IQ.DPUHLW\$GAHLW 70SO:GOLIROGO3T59YLQM14+OP\$I/XK\$M8CL6PZB*L8PK99Q9E\$BDZIF9J8-I\$GI0 J1ALL:F71APC9*KF6LF/NLR/FZ.COKEH-BB4OQ9OG4C5AO**HOELK2AZ7LBLEH-BHPLV5GK3DNKE\$JDVPLW1KD0KCZG.M1LUSB5BCQRJ\$DB5N9%V/GO4IHIBBJ-BI%NWRS%LR%\$KR46325NABFDDAFHD9PZP11COD5U*2KQXCA5W8HH/K51DQO8O0-SOSENFH9101U8$3"
+        "HC1:6BFOXN*TS0BI\$ZD.P9UOL97O4-2HH77HRM3DSPTLRR+%3QVH9M9ESIGUBA KWML:" +
```
Comment (d4rken) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/vaccination/core/VaccinationTestData.kt:142]: Single line with suppress please

**Review by d4rken — COMMENTED** [review]
lgtm, no blockers.

If possible please revert changes that convert single lines of hex/base64 to multiline and just suppress the LINT error. The multiple lines make comparison/search/copy&paste with other sources more difficult.

File: Corona-Warn-App/src/androidTest/java/testhelpers/BaseUITest.kt (hunk 5 above)
Comment (d4rken) [inline Corona-Warn-App/src/androidTest/java/testhelpers/BaseUITest.kt:20]: Did you perform a successful screenshot CI run after the UI test adjustments?

Comment (mtwalli) [reply Corona-Warn-App/src/androidTest/java/testhelpers/BaseUITest.kt:20]: I ran them locally ,but here a CI [Build](https://app.circleci.com/pipelines/github/corona-warn-app/cwa-app-android/15601/workflows/ed40c2d7-52d3-4533-a905-cfda09534041/jobs/125127)

**Review by d4rken — APPROVED** [review]
Good work! Thank you

**Review by chiljamgossow — APPROVED** [review]
Thanks for the update

**Review by SamuraiKek — APPROVED** [review]
Great work!

**Pull request merged**